crates/goose/src/agents/sub_recipe_manager.rs

```diff
@@ -0,0 +1,69 @@
+use mcp_core::{Content, Tool, ToolError};
+use serde_json::Value;
+use std::collections::HashMap;
+
+use crate::{
+    agents::recipe_tools::sub_recipe_tools::{
+        create_sub_recipe_tool, run_sub_recipe, SUB_RECIPE_TOOL_NAME_PREFIX,
+    },
+    recipe::SubRecipe,
+};
+
+#[derive(Debug, Clone)]
+pub struct SubRecipeManager {
+    pub sub_recipe_tools: HashMap<String, Tool>,
+    pub sub_recipes: HashMap<String, SubRecipe>,
+}
+
+impl Default for SubRecipeManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SubRecipeManager {
+    pub fn new() -> Self {
+        Self {
+            sub_recipe_tools: HashMap::new(),
+            sub_recipes: HashMap::new(),
+        }
+    }
+
+    pub fn add_sub_recipe_tools(&mut self, sub_recipes_to_add: Vec<SubRecipe>) {
+        for sub_recipe in sub_recipes_to_add {
+            let sub_recipe_name = sub_recipe.name.clone();
+            let tool = create_sub_recipe_tool(&sub_recipe);
+            self.sub_recipe_tools.insert(sub_recipe_name.clone(), tool);
+            self.sub_recipes.insert(sub_recipe_name, sub_recipe);
+        }
+    }
+
+    pub fn is_sub_recipe_tool(&self, tool_name: &str) -> bool {
+        tool_name.starts_with(SUB_RECIPE_TOOL_NAME_PREFIX)
```

Comment:
here you might as well just test for membership in the map, rather than the prefix